### PR TITLE
Extract wallet

### PR DESCRIPTION
**Pull request by camfairchild**
This PR is an extraction of the wallet class and related functionality to https://github.com/opentensor/bittensor-wallet as a separate python package https://pypi.org/project/bittensor-wallet/

This it to isolate contributions to wallet- and key-related functions to its own repo, where we can better monitor contributions.

### CODE REVIEW

File: bittensor/_cli/commands/inspect.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def run (cli):
         for wallet in tqdm( wallets ):
             delegates: List[Tuple(bittensor.DelegateInfo, bittensor.Balance)] = subtensor.get_delegated( coldkey_ss58=wallet.coldkeypub.ss58_address )
             if not wallet.coldkeypub_file.exists_on_device(): continue
-            cold_balance = wallet.get_balance( subtensor = subtensor )
+            cold_balance = subtensor.get_balance( wallet.coldkeypub.ss58_address )
```
Comment (ifrit98) [inline bittensor/_cli/commands/inspect.py:91]: Nice!

File: tests/unit_tests/bittensor_tests/utils/test_utils.py
Comment (ifrit98) [inline tests/unit_tests/bittensor_tests/utils/test_utils.py:1]: Think this needs to add `ddt` to `requirements/prod.txt`.

**Review by ifrit98 — CHANGES_REQUESTED** [review]
Everything looks good.

One nit: can you add `ddt==1.6.0` to the `requirements/prod.txt` so the `test_utils.py` passes?

**Comment by camfairchild** [comment]
> Everything looks good.
> 
> One nit: can you add `ddt==1.6.0` to the `requirements/prod.txt` so the `test_utils.py` passes?

That requirement is in `requirements/dev.txt` with the other test requirements. 
You can do `pip install -e .[dev]` or `pip install -e '.dev'` on Mac to install the dev extra

File: tests/unit_tests/bittensor_tests/utils/test_utils.py
Comment (camfairchild) [reply tests/unit_tests/bittensor_tests/utils/test_utils.py:1]: https://github.com/opentensor/bittensor/pull/1403#issuecomment-1610589565

**Review by ifrit98 — APPROVED** [review]
Wonderful, 🙏

**Pull request merged**